src/OrleansAWSUtils/Streams/SQSBatchContainer.cs

```diff
@@ -0,0 +1,112 @@
+﻿using Amazon.SQS.Model;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Orleans.Providers.Streams.Common;
+using Orleans.Runtime;
+using Orleans.Serialization;
+using Orleans.Streams;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using SQSMessage = Amazon.SQS.Model.Message;
+
+namespace OrleansAWSUtils.Streams
+{
+    [Serializable]
+    internal class SQSBatchContainer : IBatchContainer
```

Comment:
I'll do it myself. Just need to get this AWS package shipped and will get this refactor. It will probably introduce a common class in the core and those providers will just pass a backend implementation on a single interface. All this stream and Storage providers looks like the same code to me.